grafana-dashboard.tf

```diff
@@ -0,0 +1,36 @@
+resource "random_integer" "graph_id" {
+  min     = 1000
+  max     = 999999
+  count = length(local.apiary_data_buckets)
+}
```

Comment:
That's true! I originally had a different solution that would increment the integer by one for every new graph. However, this is cumbersome since we are creating multiple graphs per bucket so we would have to change our tf for loop with the amount of graphs we want to create every time we add a graph.

uuid would not work for this field as it needs to be an int. However, I it seems grafana may make it's own ID if this field is left null. Testing this now.

Reply:
If you leave graph_id NULL, you can delete this `random_integer` resource.

Reply:
That didn't work. If we don't add an ID, grafana will not display the dashboard at all. I tried a few solutions and kept getting errors. 

So, I reverted to the original solution I had, basically we will start with integer 100 for bucket graph ids (because we would like to allow graphs that are not created once per bucket but are general for all buckets which will have ids starting with 1) and will increment it by one for every other graph id. 

However, the major issue with this solution is that if we wanted to add a third graph per bucket to our dashboard, we will need to change not only the template file, but also the terraform itself:

in grafana-dashboard.tf: 
`    graph_id = range(100, length(local.apiary_data_buckets) * 2, 2)[count.index]` 

The reason why we multiply the length of apiary data buckets by two and skip every other integer is because we have two graphs per bucket in this dashboard. The template has these two ids for the two graphs: 

`      "id": ${graph_id},`
`      "id": ${graph_id+1},`
 
If we wanted to add a new graph to this dashboard, in the template it will have `"id": ${graph_id+2},` and in the tf we will have to change the for loop to be `graph_id = range(100, length(local.apiary_data_buckets) * 3, 3)[count.index]`

I originally thought this solution was a bit harder to understand and updating in the future would not be as straight forward. However, I also understand the point that the random integer can repeat itself which will make the dashboard miss one or more graphs. 

So, what do you think is better, random integer or increasing for loop? Do you have a better solution for this?

Reply:
I think the for loop is the best solution - I certainly don't have a better one.  However, this probably needs documenting in the README, and instead of hardcoding `100` and `2`, you could define `local` variables at the top of `grafana_dashboard.tf` like:
```
locals {
   graph_id_base                        = "100"
   num_s3_graphs_per_bucket = "2"
}
```
so that it is easy to find and modify those two values.